sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -175,16 +191,28 @@ def plot_precision_recall_curve(estimator, X, y, *,
         raise ValueError(classification_error)
 
     prediction_method = _check_classifier_response_method(estimator,
-                                                         response_method)
+                                                          response_method)
     y_pred = prediction_method(X)
 
-    if y_pred.ndim != 1:
+    if pos_label is not None and pos_label not in estimator.classes_:
+        raise ValueError(
+            f"The class provided by 'pos_label' is unknown. Got "
+            f"{pos_label} instead of one of {estimator.classes_}"
+        )
+
+    if y_pred.ndim != 1:  # `predict_proba`
         if y_pred.shape[1] != 2:
             raise ValueError(classification_error)
-        else:
+        if pos_label is None:
+            pos_label = estimator.classes_[1]
             y_pred = y_pred[:, 1]
+        else:
+            class_idx = np.flatnonzero(estimator.classes_ == pos_label)
+            y_pred = y_pred[:, class_idx]
+    else:  # `decision_function`
+        if pos_label is None:
+            pos_label = estimator.classes_[1]
```

Comment:
Do we invert the `y_pred` when the `pos_label == estimator.classes_[0]`? Concretely:

```py
if pos_label == estimator.classes_[0]:
    y_pred *= -1
```

The `decision_function` is associated with `estimator.classes_[1]` and `precision_recall_curve` will mark `estimator.classes_[0]` as the positive for `y_true`.

Reply:
It should be done in the precision/recall function otherwise my test would not pass, isn't it?

Reply:
Your test pass with or without the inversion.

Reply:
both

Reply:
I will check tomorrow the reason :)

Reply:
You are right. Only the computation of the accuracy score was fine which I used in my test.
Then I will also improve my test :)

Reply:
OK I added a test which should give an estimate of the average precision by integrating on the precision-recall array and check that they are in line with the computed average precision. Second thing is that I manually set the bound to be sure that we are in a specific setting of `pos_label`